app/assets/javascripts/dashboard/dashboard.js

```diff
@@ -561,10 +562,12 @@ function healthCheckToReload() {
 // begin unassigned
 function isAssignable() {
   var errors = [];
+  var masters = State.assignedMinions.filter(function (m) { return m.role === 'master' });
 
   // We need an odd number of masters
-  if (selectedMastersLength() % 2 !== 0) {
-    errors.push('The number of masters to be added has to be an even number in order to maintain the odd constraint number in the cluster');
+  if ((masters.length + selectedMastersLength()) % 2 === 0 &&
+      selectedMastersLength() > 0) {
```

Comment:
for optimization i would switch the logic here to first check for `selectedMastersLength() > 0` - this saves the heavyer calculation part in case `selectedMastersLength()` is 0

Reply:
Not a big deal during these days but you are right. I'll change it. Thanks!

Reply:
Done!